pkg/providers/vsphere/validator.go

```diff
@@ -89,7 +89,31 @@ func (v *Validator) ValidateVCenterConfig(ctx context.Context, datacenterConfig
 	return nil
 }
 
-// TODO: dry out machine configs validations.
+func (v *Validator) validateMachineConfigTagsExist(ctx context.Context, machineConfigs []*anywherev1.VSphereMachineConfig) error {
+	tags, err := v.govc.ListTags(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to check if tags exists in vSphere: %v", err)
+	}
+
+	tagIds := make([]string, 0, len(tags))
+	for _, t := range tags {
+		tagIds = append(tagIds, t.Id)
+	}
+
+	idLookup := types.SliceToLookup(tagIds)
+	for _, machineConfig := range machineConfigs {
+		for _, tagID := range machineConfig.Spec.TagIDs {
+			if !idLookup.IsPresent(tagID) {
+				return fmt.Errorf("tag (%s) does not exist in vSphere. please provide a valid tag id in the urn format (example: urn:vmomi:InventoryServiceTag:8e0ce079-0677-48d6-8865-19ada4e6dabd:GLOBAL)", tagID)
+			}
+		}
+	}
+
+	logger.MarkPass("Machine config tags validated")
```

Comment:
should there be a space after this logger line? looks like most of the other funcs have a space before the final return.